Add validations to listings

Thrillberg opened the pull request.

Thrillberg commented:
This is a first pass at adding validations but I have a couple questions:
1) Did I require the correct fields? Any that we want to allow to be `nil`?
2) I did not add a migration to make the validations enforced on the database, but should we?
3) I added tests for them but tests are missing for other things the model does. Should we add those? (specifically, I'm thinking of the `belongs_to :creator`, `has_many_attached :photos`, and `acts_as_taggable_on :tags` lines) If we want to add those, perhaps under a separate commit would be a good idea.
4) Do we want to add feature tests?

maebeale commented:
@Thrillberg 

1. Model validations: 
    - description can be nil
    - reference_url can be nil
2. Probably we should, yes. I generally prefer required validations to be in both places, but open to your suggestion.
3. Sounds great!
4. Would love them!

Thrillberg committed:
Add validations for Listings

This commit adds presence validations for some fields on the Listing
model. Issue #147 digs into a question raised by this commit: do we want
to enforce presence of these fields on the database, too?

Fixess #138

Thrillberg commented:
@maebeale I've updated this branch with code that I _think_ is ready for review. It spun off a few other issues based on my points above. For point 4 above, I'm not sure how feature tests are currently being handled in the app so I'll hold off on that for now.
